sockeye/beam_search.py

```diff
@@ -674,7 +674,7 @@ def forward(self,
                 break
 
             # (5) update models' state with winning hypotheses (ascending)
-            _sort_states(model_states, best_hyp_indices)
+            model_states = _sort_states(model_states, best_hyp_indices)
```

Comment:
This was the bug.

Reply:
👏👏 great catch! Thanks!

Reply:
👏👏 great catch! Thanks!

Reply:
Took a while to debug because I comparing the model states by taking the sum over ndarrays which matched even when entries were reshuffled. Anyway, good to have the refactoring back. Having beam search separated from the rest of inference looks much cleaner!